### PR TITLE
[SPARK-46171][SQL][PYTHON][R][DOCS] Remove `HiveContext` from Apache Spark 4.0

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr aims to  remove the deprecated  `HiveContext` from Apache Spark 4.0 since `HiveContext` has been marked as deprecated after Spark 2.0.

### Why are the changes needed?
Clean up the long-deprecated API.

### Does this PR introduce _any_ user-facing change?
Yes, there is no longer `HiveContext` available, users need to use `SparkSession` with `enableHiveSupport`

### How was this patch tested?
Pass GitHub Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: python/pyspark/sql/context.py [hunk 1]
```diff
@@ -311,6 +312,24 @@ def registerJavaFunction(
         )
         return self.sparkSession.udf.registerJavaFunction(name, javaClassName, returnType)
 
+    # TODO(andrew): delete this once we refactor things to take in SparkSession
+    def _inferSchema(self, rdd: RDD, samplingRatio: Optional[float] = None) -> StructType:
```
Comment (dongjoon-hyun) [inline python/pyspark/sql/context.py:316]: What is this addition?

Comment (LuciferYang) [reply python/pyspark/sql/context.py:316]: Sorry, I think I made a mistake during the rebase.

**Comment by dongjoon-hyun** [comment]
I agree with you, @LuciferYang . Thank you for the heads-up email.
Since we need an official vote for this specific API, I replied your email thread.
Let's see the community atmosphere.

File: python/pyspark/sql/context.py (hunk 1 above)
Comment (LuciferYang) [reply python/pyspark/sql/context.py]: Thanks @dongjoon-hyun, the code in this file should be okay now.

**Review by gatorsmile — CHANGES_REQUESTED** [review]
It sounds like the maintenance cost is not high. However, removal will make it difficult for our end users to migrate from the old release to version 4.0.

**Review by gatorsmile — CHANGES_REQUESTED** [review]
It sounds like the maintenance cost is not high. However, removal will make it difficult for our end users to migrate from the old release to version 4.0.

**Review by gatorsmile — CHANGES_REQUESTED** [review]
It sounds like the maintenance cost is not high. However, removal will make it difficult for our end users to migrate from the old release to version 4.0.

**Comment by LuciferYang** [comment]
After carefully reading  https://lists.apache.org/thread/mrx0y078cf3ozs7czykvv864y6dr55xq, I have decided to abandon the deletion of HiveContext. As @gatorsmile said, its maintenance cost is not high, but it will increase the cost of users migrating to Spark4.0, so I also believe it is not worth deleting it in this context.

I will close this pr. Thanks @dongjoon-hyun @gatorsmile

**Comment by dongjoon-hyun** [comment]
Thank you for your decision, @LuciferYang .

**Pull request closed without merging**